include/deal.II/fe/mapping_q_generic.h

```diff
@@ -524,29 +571,62 @@ class MappingQGeneric : public Mapping<dim,spacedim>
-   * points on the perimeter of a quad to get the location of interior support
-   * points.
+   * A vector of tables of weights by which we multiply the locations of the
+   * support points on the perimeter of an object (line, quad, hex) to get the
+   * location of interior support points.
+   *
+   * Access into this table is by @p [structdim-1], i.e., use 0 to access the
+   * support point weights on a line (i.e., the interior points of the
+   * GaussLobatto quadrature), use 1 to access the support point weights from
+   * to perimeter to the interior of a quad, and use 2 to access the support
+   * point weights from the perimeter to the interior of a hex.
    *
-   * Sizes: support_point_weights_on_quad.size()= number of inner
-   * unit_support_points support_point_weights_on_quad[i].size()= number of
-   * outer unit_support_points, i.e.  unit_support_points on the boundary of
-   * the quad
+   * The table itself contains as many columns as there are surrounding points
+   * to a particular object (2 for a vertex, <code>4 + 4*(degree-1)</code> for
+   * a quad, <code>8 + 12*(degree-1) + 6*(degree-1)*(degree-1)</code> for a
+   * hex) and as many rows as there are strictly interior points.
    *
-   * For the definition of this vector see equation (8) of the `mapping'
+   * For the definition of this table see equation (8) of the `mapping'
    * report.
    */
-  Table<2,double> support_point_weights_on_quad;
+  std::vector<Table<2,double> > support_point_weights_perimeter_to_interior;
 
   /**
-   * A table of weights by which we multiply the locations of the support
-   * points on the perimeter of a hex to get the location of interior support
-   * points.
+   * A vector of tables of weights by which we multiply the locations of the
+   * vertex points of an object (line, quad, hex) to get the location of
+   * interior support points.
    *
-   * For the definition of this vector see equation (8) of the `mapping'
-   * report.
+   * As opposed to @p support_point_weights_perimeter_to_interior, this table
+   * takes only the vertex points into account and not intermediate points
+   * inside the objects. Thus, the content of this table simply corresponds to
+   * the evaluation of the linear shape functions in the dimension of the
+   * object.
+   *
+   * Access into the vector of tables is by @p [structdim-1], i.e., use 0 to
+   * access the support point weights on a line (i.e., the interior points of
+   * the GaussLobatto quadrature), use 1 to access the support point weights
+   * from to perimeter to the interior of a quad, and use 2 to access the
+   * support point weights from the perimeter to the interior of a hex.
+   *
+   * The table itself contains as many columns as there are vertices on a
+   * particular object (2 for a vertex, 4 for a quad, 8 for a hex) and as many
+   * rows as there are strictly interior points.
+   */
+  std::vector<Table<2,double> > support_point_weights_interior;
```

Comment:
Is it worth adding to the documentation that this table is only used if the smoothing is off?